api-demo/src/advancedDemo.ts

```diff
@@ -0,0 +1,163 @@
+    comments.push(...ts.getTrailingCommentRanges(buffer, node.getFullStart()) || []);
+
+    if (comments.length > 0) {
+      console.log(indent + colors.yellow('  FOUND COMMENT'));
+      const comment: ts.CommentRange = comments[0];
+      visitorContext.commentNode = node;
+      visitorContext.commentText = tsdoc.TextRange.fromStringRange(buffer, comment.pos, comment.end);
+    }
+  }
+
+  return node.forEachChild(child => visitCompilerAst(child, indent + '  ', visitorContext));
+}
+
+function parseTSDoc(textRange: tsdoc.TextRange, commentNode: ts.Node): void {
+  console.log(os.EOL + colors.green('Input Buffer:') + os.EOL);
+  console.log(colors.gray('<<<<<<'));
+  console.log(textRange.toString());
+  console.log(colors.gray('>>>>>>'));
+
+  const customConfiguration: tsdoc.TSDocParserConfiguration = new tsdoc.TSDocParserConfiguration();
+
+  const customInlineDefinition: tsdoc.TSDocTagDefinition = new tsdoc.TSDocTagDefinition({
+    tagName: '@customInline',
+    syntaxKind: tsdoc.TSDocTagSyntaxKind.InlineTag,
+    allowMultiple: true
+  });
+
+  // NOTE: Defining this causes a new DocBlock to be created under docComment.customBlocks.
+  // Otherwise, a simple DocBlockTag would appear inline in the @remarks section.
+  const customBlockDefinition: tsdoc.TSDocTagDefinition = new tsdoc.TSDocTagDefinition({
+    tagName: '@customBlock',
+    syntaxKind: tsdoc.TSDocTagSyntaxKind.BlockTag
+  });
+
+  // NOTE: Defining this causes @customModifier to be removed from its section,
+  // and added to the docComment.modifierTagSet
+  const customModifierDefinition: tsdoc.TSDocTagDefinition = new tsdoc.TSDocTagDefinition({
+    tagName: '@customModifier',
+    syntaxKind: tsdoc.TSDocTagSyntaxKind.ModifierTag
+  });
+
+  customConfiguration.addTagDefinitions([
+    customInlineDefinition,
+    customBlockDefinition,
+    customModifierDefinition
+  ]);
+
+  console.log(os.EOL + 'Invoking TSDocParser with custom configuration...' + os.EOL);
+  const tsdocParser: tsdoc.TSDocParser = new tsdoc.TSDocParser(customConfiguration);
+  const parserContext: tsdoc.ParserContext = tsdocParser.parseRange(textRange);
+  const docComment: tsdoc.DocComment = parserContext.docComment;
+
+  console.log(os.EOL + colors.green('Parser Log Messages:') + os.EOL);
+
+  if (parserContext.log.messages.length === 0) {
+    console.log('No errors or warnings.');
+  } else {
+    const sourceFile: ts.SourceFile = commentNode.getSourceFile();
+    for (const message of parserContext.log.messages) {
+      // Since we have the compiler's analysis, use it to calculate the line/column information,
+      // since this is currently faster than TSDoc's TextRange.getLocation() lookup.
+      const location: ts.LineAndCharacter = sourceFile.getLineAndCharacterOfPosition(message.textRange.pos);
+      const formattedMessage: string = `${sourceFile.fileName}(${location.line + 1},${location.character + 1}):`
+        + ` [TSDoc] ${message}`;
```

Comment:
Do you have a function for this?

Reply:
Possibly the TypeScript compiler has an API for this, but I think it's better to be explicit for the demo.

TSDoc does this formatting, but as noted it's more efficient to use the compiler here.

---
In reply to: [214224647](https://github.com/Microsoft/tsdoc/pull/55#discussion_r214224647) [](ancestors = 214224647)